### PR TITLE
Handle Null case for triggers in prod JSON

### PR DESCRIPTION
**Pull request by bfontecc007**
Signed-off-by: Bret Fontecchio <bfontecc@redhat.com>

### CODE REVIEW

File: tests/build_/test_build_request.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
                              TEST_COMPONENT, TEST_GIT_BRANCH, TEST_GIT_REF,
                              TEST_GIT_URI, TEST_GIT_URI_HUMAN_NAME)
 
+USE_DEFAULT_TRIGGERS = object()
```
Comment (twaugh) [inline tests/build_/test_build_request.py]: Extra blank line needed here for PEP8.
```
tests/build_/test_build_request.py:34:1: E302 expected 2 blank lines, found 1
```

Comment (lcarva) [inline tests/build_/test_build_request.py]: I wonder why stickler didn't complain about this.

**Comment by twaugh** [comment]
coveralls is reporting a decrease in coverage because now https://github.com/projectatomic/osbs-client/pull/600/files#diff-6451bca7357ec20f16d74cb3cc6d5553R175 never gets control.

Maybe we need a test case for 'there are triggers, but none are imageChange'? e.g. one of these: https://docs.openshift.org/latest/dev_guide/builds/triggering_builds.html#generic-webhooks

File: osbs/build/build_request.py [hunk 2]
```diff
@@ -166,6 +166,8 @@ def dj(self):
     def has_ist_trigger(self):
         """Return True if this BuildConfig has ImageStreamTag trigger."""
         triggers = self.template['spec'].get('triggers', [])
+        if not triggers:
```
Comment (lcarva) [inline osbs/build/build_request.py]: To address the code coverage decrease, we could remove this if-block and change the for loop to:
```python
for trigger in triggers or []:
```

Comment (lcarva) [inline osbs/build/build_request.py]: Or maybe better yet to set it to an empty list when triggers variable is defined:
```python
triggers = self.template['spec'].get('triggers') or []
```

In either case, this is fine as is to me.

**Comment by bfontecc007** [comment]
Looks like I got coverage back up
